android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -267,9 +267,11 @@ protected void onResume() {
 
     // onConfigurationChanged() only triggers when device is rotated while app is in foreground
     // This handles when device is rotated while app is in background
+    // using KMManager.getOrientation() since getConfiguration().orientation is unreliable #10241
     Configuration newConfig = this.getResources().getConfiguration();
-    if (newConfig != null && newConfig.orientation != lastOrientation) {
-      lastOrientation = newConfig.orientation;
+    int newOrientation = KMManager.getOrientation(context);
+    if (newOrientation != lastOrientation) {
+      lastOrientation = newOrientation;
       KMManager.onConfigurationChanged(newConfig);
     }
```

Comment:
This retrieves `newConfig` but never tests it. Given it may have a wrong orientation, is this going to cause grief when passing it to `onConfigurationChanged()`.
Also, I would move line 271 `... newConfig = ...` inside the `if() {` statement before line 275 in order to keep its scope limited

Reply:
I retested the scenarios from #8704 and they seemed fine (restarting the app after rotating the device while the app is in the background).
lastOrientation is undefined so the app goes into this branch and refreshes with the correct dimensions. Maybe getOrientation() is the only buggy part?